provider/plugin/plugin.go

```diff
@@ -0,0 +1,350 @@
+		return nil, err
+	}
+	req, err := http.NewRequest("GET", u, nil)
+	if err != nil {
+		recordsErrorsGauge.Inc()
+		log.Debugf("Failed to create request: %s", err.Error())
+		return nil, err
+	}
+	req.Header.Set(acceptHeader, mediaTypeFormatAndVersion)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		recordsErrorsGauge.Inc()
+		log.Debugf("Failed to perform request: %s", err.Error())
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		recordsErrorsGauge.Inc()
+		log.Debugf("Failed to get records with code %d", resp.StatusCode)
+		return nil, fmt.Errorf("failed to get records with code %d", resp.StatusCode)
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		recordsErrorsGauge.Inc()
+		log.Debugf("Failed to read response body: %s", err.Error())
+		return nil, err
+	}
+
+	endpoints := []*endpoint.Endpoint{}
+	err = json.Unmarshal(b, &endpoints)
+	if err != nil {
+		recordsErrorsGauge.Inc()
+		log.Debugf("Failed to unmarshal response body: %s", err.Error())
+		return nil, err
+	}
+	return endpoints, nil
+}
+
+// ApplyChanges will make a POST to remoteServerURL/records with the changes
+func (p PluginProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) error {
+	u, err := url.JoinPath(p.remoteServerURL.String(), "records")
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("Failed to join path: %s", err.Error())
+		return err
+	}
+	b, err := json.Marshal(changes)
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("Failed to marshal changes: %s", err.Error())
+		return err
+	}
+
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("Failed to create request: %s", err.Error())
+		return err
+	}
+	req.Header.Set(acceptHeader, mediaTypeFormatAndVersion)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("Failed to perform request: %s", err.Error())
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		applyChangesErrorsGauge.Inc()
+		log.Debugf("Failed to apply changes with code %d", resp.StatusCode)
+		return fmt.Errorf("failed to apply changes with code %d", resp.StatusCode)
+	}
+	return nil
+}
+
+// PropertyValuesEqual will call the provider doing a POST on `/propertyvaluesequal` which will return a boolean in the format
+// `{propertyvaluesequal: true}`
+// Errors in anything technically happening from the provider will return true so that no update is performed.
+// Errors will also be logged and exposed as metrics so that it is possible to alert on them if needed.
+func (p PluginProvider) PropertyValuesEqual(name string, previous string, current string) bool {
+	u, err := url.JoinPath(p.remoteServerURL.String(), "propertyvaluesequal")
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to join path: %s", err)
+		return true
+	}
```

Comment:
```suggestion
        u := p.remoteServerURL.JoinPath("propertyvaluesequal").String()
```
see [this comment](https://github.com/kubernetes-sigs/external-dns/pull/3063/files/bdc19db88b1a9793cc2e2473e1c932673709112d#r1224361705)